src/ui/components/ExperimentBrowserPage.tsx

```diff
@@ -60,67 +60,73 @@ const ExperimentBrowserPage: FC = () => {
+            addToast({ message: "Enrollment successful", variant: "success" });
+          } else {
+            addToast({ message: "Enrollment failed", variant: "danger" });
+          }
+        } catch (error) {
+          addToast({
+            message: `Error enrolling into experiment: ${(error as Error).message ?? String(error)}`,
+            variant: "danger",
+          });
         }
-      } catch (error) {
+      } else {
         addToast({
-          message: `Error enrolling into experiment: ${(error as Error).message ?? String(error)}`,
+          message: "Select a branch before enrolling",
           variant: "danger",
         });
       }
-    } else {
-      addToast({
-        message: "Select a branch before enrolling",
-        variant: "danger",
-      });
-    }
-  };
+    },
+    [experiments, addToast],
+  );
 
-  const handleGenerateTestIds = async (
-    experimentId: string,
-    branchSlug: string,
-  ) => {
-    if (branchSlug) {
-      const recipe = experiments.find((exp) => exp.id === experimentId);
-      try {
-        const result = await browser.experiments.nimbus.generateTestIds(
-          recipe,
-          branchSlug,
-        );
-        if (result) {
-          await navigator.clipboard.writeText(result);
+  const handleGenerateTestIds = useCallback(
+    async (experimentId: string, branchSlug: string) => {
+      if (branchSlug) {
+        const recipe = experiments.find((exp) => exp.id === experimentId);
+        try {
+          const result = await browser.experiments.nimbus.generateTestIds(
+            recipe,
+            branchSlug,
+          );
+          if (result) {
+            await navigator.clipboard.writeText(result);
+            addToast({
+              message: `Id successfully generated and copied to clipboard. Test Id: ${result}`,
+              variant: "success",
+              autohide: false,
+            });
+          } else {
+            addToast({
+              message: "Test Id generation failed",
+              variant: "danger",
+            });
+          }
+        } catch (error) {
           addToast({
-            message: `Id successfully generated and copied to clipboard. Test Id: ${result}`,
-            variant: "success",
-            autohide: false,
+            message: `Error generating test Id: ${(error as Error).message ?? String(error)}`,
+            variant: "danger",
           });
-        } else {
-          addToast({ message: "Test Id generation failed", variant: "danger" });
         }
-      } catch (error) {
+      } else {
         addToast({
-          message: `Error generating test Id: ${(error as Error).message ?? String(error)}`,
+          message: "Select a branch before generating test Id",
           variant: "danger",
         });
       }
-    } else {
-      addToast({
-        message: "Select a branch before generating test Id",
-        variant: "danger",
-      });
-    }
-  };
+    },
+    [experiments, addToast],
```

Comment:
We can also lift this up to a top-level function that takes `experiment` and returns the value to pass to addToast